src/name/ip_address.rs

```diff
@@ -278,7 +278,7 @@ pub(super) fn presented_id_matches_constraint(
         return Err(Error::BadDER);
     }
     if constraint.len() != 8 && constraint.len() != 32 {
-        return Err(Error::BadDER);
+        return Err(Error::InvalidNetworkMaskConstraint);
```

Comment:
Should we do a similar match here as you did elsewhere that more precisely lists the valid length combinations, replacing a few different `if`s at once?

Reply:
added in ae25632